dandiapi/api/management/commands/delete_malformed_manifests.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+
+import logging
+
+from django.core.files.storage import default_storage
+import djclick as click
+from tqdm import tqdm
+
+from dandiapi.api.manifests import _manifests_path, all_manifest_filepaths
+from dandiapi.api.models import Version
+from dandiapi.api.storage import get_boto_client
+from dandiapi.api.tasks import write_manifest_files
+
+logger = logging.getLogger(__name__)
+
+
+def list_files_in_manifest_directory(version: Version) -> list[str]:
+    """List all files in the manifest directory for a version."""
+    manifest_dir = _manifests_path(version)
+    client = get_boto_client()
+
+    # List objects with the manifest directory prefix
+    paginator = client.get_paginator('list_objects_v2')
+    pages = paginator.paginate(
+        Bucket=default_storage.bucket_name, Prefix=manifest_dir + '/', Delimiter='/'
+    )
+
+    return [obj.get('Key', '') for page in pages for obj in page.get('Contents', [])]
```

Comment:
Since the value returned from this function is only used once to build up another list, we should be able to make this a lazy iterator instead -
```suggestion
def list_files_in_manifest_directory(version: Version) -> Iterator[str]:
    """List all files in the manifest directory for a version."""
    manifest_dir = _manifests_path(version)
    client = get_boto_client()

    # List objects with the manifest directory prefix
    paginator = client.get_paginator('list_objects_v2')
    pages = paginator.paginate(
        Bucket=default_storage.bucket_name, Prefix=manifest_dir + '/', Delimiter='/'
    )

    for page in pages for obj in page.get('Contents', []):
      if 'Key' in obj:
        yield obj['Key']
```